src/callbacks/callback_ltfb.cpp

```diff
@@ -91,8 +90,27 @@ void exchange_weights(lbann_comm* comm,
       }
     }
     if (size > 0 && send) {
-      comm->sendrecv(local_matrix.LockedBuffer(), size, partner,
-                     remote_matrix->Buffer(), size, partner);
+      switch (remote_matrix->GetLocalDevice()) {
+      case El::Device::CPU:
+        comm->sendrecv(local_matrix.LockedBuffer(), size, partner,
+                       remote_matrix->Buffer(), size, partner,
+                       El::SyncInfo<El::Device::CPU>{});
+        break;
+#ifdef HYDROGEN_HAVE_CUDA
+      case El::Device::GPU:
+        using ValueT
+          = std::remove_pointer<decltype(remote_matrix->Buffer())>::type;
+        comm->sendrecv(
+          local_matrix.LockedBuffer(), size, partner,
+          remote_matrix->Buffer(), size, partner,
+          El::SyncInfo<El::Device::GPU>{
+            static_cast<El::Matrix<ValueT,El::Device::GPU> const&>(
```

Comment:
This is very general, although maybe overkill. `static_cast<GPUMat const&>` would probably be sufficient.